service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/batch/ResponseMerger.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2015 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.poseidon.serviceclients.batch;
+
+import java.util.List;
+
+/**
+ * Created by venkata.lakshmi on 16/04/15.
+ */
+public interface ResponseMerger<T> {
+
+    public T mergeResponse(List<T> responses);
+}
```

Comment:
Please see how we can automatically split/merge array and collection types. Most likely we should provide concrete implementations of these two interfaces for iterable types like any array of objects as well as any collection of objects which would get widely re-used. Only in very few corner cases, users would have to write new splitter and merger classes.